### ISSUE TITLE
How do you run this script?

### ISSUE THREAD

**Issue by PyroTeck** [Seklfreak/discord-image-downloader-go#15]
You have a section on how to run it, but I don't know Go programming. I've installed it onto my computer, but I have no idea how to run the script itself.

**Comment by xExoTicxLeGeNDx** [comment]
Yeah, I'm also having a problem with this the section that tells you how to use is a bit out there doesn't help in the slightest.

**Comment by DreamPhreak** [comment]
You don't need to know Go language to use it (I don't know it either).

After you download it, move the .exe file into it's own folder, then run it once. It will generate the files and then close itself to give you a chance to edit the settings. 
Open config.ini, fill in the [auth] section with your discord account (if you want to set up a dedicated discord-downloader that runs 24/7, it might be better to make a 2nd discord account and use that info instead if your main account)
The next section, [general] settings: skip edits, if someone edits their comments with new pics/links, if you want to ignore that or not. (personally I set it to false to get the edits too) - download tistory sites, if you dont know what this is, just leave it false - max download retries: if a download fails, it will keep trying this amount of times (personally I lowered it to 1 or 2 because 5 retries takes a long time) - download timeout: how long of trying to download before it gives up - send notices: might become spammy and get banned from servers if you set this to true to let the channels know you're downloading.

[channels] section, the account must be in the server for it to download from channels. (Recommended you do this from the browser version of Discord to see URLs) Lets say you join a server about sharing Cat pics, one of the channels is for fluffy kittens. You click on the #channel and in the URL it will say like discordapp.com/channels/SERVERID/CHANNELID - just copy/paste the CHANNELID into the file where it says "channelid1 =" and then after replace the "C:\full\path\1" with wherever on your system you want to save the pics, like for example "C:\Users\JohnDoe\Downloads\FluffyKittens", whatever you want. (Personally I also organize by server and then channel.)

If you're setting up a dedicated downloader, you should add a new section to the bottom of the file:
```
[interactive channels]
<your channel id> = <some valid path>
```
This part gets a bit messy and maybe confusing for new users. With the browser-based discord still open, you'll have to find a way to send a message to the account, maybe open up the discord client or use a 2nd computer, either of which logged into your main account. Send a message to the 2nd account just to start the chat. On the 2nd account in the browser, in the URL for that private message it should be like discordapp.com/channels/@me/CHANNELID - Copy/paste that CHANNELID into the <your channel id> section (no brackets). The <some valid path> doesnt matter, I don't think it does anything, just a path to wherever on your computer that's valid.

Save the config.ini. Run the .exe. Now with your main account, you can send the command "history" by itself to the 2nd account in a private message, then it will ask which CHANNELID do you want to download the entire history of. The only valid CHANNELIDs are the ones you set in the config.ini in the [channels] section. Just send the CHANNELID by itself and it will start downloading.

Unfortunately, if you're like me and set up 50+ channels, you'll have to do those last steps for every single channel that you set up. I wish there was a way to just dump a list of channels and have it go through each one so it can work on it's own and I won't have to be babysitting it...

Hope this helps. Sorry Seklfreak for contributing to necro issue. (March 10th)

**Comment by Seklfreak** [comment]
@DreamPhreak Wow, thank you so much for this extensive guide!

**Comment by xExoTicxLeGeNDx** [comment]
@DreamPhreak Thanks for clearing that up much appreciated

**Comment by Zwizzor** [comment]
Would it work with WINE on linux?

**Comment by Seklfreak** [comment]
Maybe? I would recommend just building a native linux executable yourself.

**Comment by Zwizzor** [comment]
I have no idea on how to make that. Also would the config.ini paths still work with linux paths?

**Comment by Seklfreak** [comment]
Linux paths should work

**Comment by Xanthraxxx** [comment]
Does the secondary option work with downloading images off the second account? Sent help in a DM and never get a popup telling me for an id.

**Comment by thibaut3000** [comment]
i don't have any .exe in the file i downloaded

**Comment by yung-feli** [comment]
Ok I did everything the instructions had me do but the bot doesnt send any message for channel ID or anything. I have a pic of my settings bellow. What do I do!!?? Also, am I supposed to close the cmd after it has logged into the account? I'm just so confused

![capture](https://user-images.githubusercontent.com/36191620/35855403-16cf39e6-0b6e-11e8-9344-72d539021dc8.PNG)

**Comment by chrtphantom** [comment]
Don't understand for download history

**Comment by Makein** [comment]
It doesnt work for me... Here:
 discord-image-downloader-go version 1.29.3
Client is now connected as Makein. Press CTRL-C to exit.


```
[auth]
token = ******************

[general]
skip edits                           = true
download tistory sites               = false
max download retries                 = 5
download timeout                     = 60
send notices to interactive channels = false

[channels]
channelid1 = 437254129400938498 E:\3\db
437254129400938498 = E:\3\db
437254129400938498 E:\3\db

[flickr]
api key = your flickr api key

[twitter]
consumer key        = your consumer key
consumer secret     = your consumer secret
access token        = your access token
access token secret = your access token secret

[interactive channels]
<your channel id> = <some valid path>
channelid4 = 437254129400938498 E:\3\db
```

**Comment by CritFlaw** [comment]
Has this become obselete? I got it working last year but nothing I do now seems to work.

**Comment by HeylonNHP** [comment]
What is the history command and how do I run it?

**Comment by hvbibi** [comment]
Hello friends! Can I ask for some help? I've done and followed everything but my selfbot account doesn't respond to the history command? here is the config. I do not know if I have done anything wrong, but I've been at it for about an hour and a half now.

[auth]
token = *****************************************

[general]
skip edits                           = true
download tistory sites               = false
max download retries                 = 5
download timeout                     = 60
send notices to interactive channels = false

[channels]
482761204868317199 = C:\Users\Wanna\Pictures\LOONA\LOML2
channelid2 = C:\full\path\2
channelid3 = C:\full\path\3

[flickr]
api key = your flickr api key

[twitter]
consumer key        = your consumer key
consumer secret     = your consumer secret
access token        = your access token
access token secret = your access token secret

[interactive channel]
590510712711807007 = C:\Users\Wanna\Pictures\LOONA\LOML2

**Comment by mtthwmndza** [comment]
> Hello friends! Can I ask for some help? I've done and followed everything but my selfbot account doesn't respond to the history command? here is the config. I do not know if I have done anything wrong, but I've been at it for about an hour and a half now.
> 
> [auth]
> token = *****************************************
> 
> [general]
> skip edits = true
> download tistory sites = false
> max download retries = 5
> download timeout = 60
> send notices to interactive channels = false
> 
> [channels]
> 482761204868317199 = C:\Users\Wanna\Pictures\LOONA\LOML2
> channelid2 = C:\full\path\2
> channelid3 = C:\full\path\3
> 
> [flickr]
> api key = your flickr api key
> 
> [twitter]
> consumer key = your consumer key
> consumer secret = your consumer secret
> access token = your access token
> access token secret = your access token secret
> 
> [interactive channel]
> 590510712711807007 = C:\Users\Wanna\Pictures\LOONA\LOML2

don't close the cmd exe. try sending history while the exe is still open.
